### PR TITLE
Add distributed hypertable support to hypertable_approximate_row_count()

### PR DESCRIPTION
**Pull request by pmwkaa**
Related to: https://github.com/timescale/timescaledb/issues/1902

### CODE REVIEW

File: sql/size_utils.sql [hunk 1]
```diff
@@ -365,6 +365,76 @@ BEGIN
...
+$BODY$
+DECLARE
+BEGIN
+	RETURN QUERY
+	SELECT h.schema_name,
+		h.table_name,
+		row_estimate.row_estimate
+	FROM _timescaledb_catalog.hypertable h
+	CROSS JOIN LATERAL (
+		SELECT sum(cl.reltuples)::BIGINT AS row_estimate
```
Comment (gayyappan) [inline sql/size_utils.sql:427]: We have to take compressed chunks into account. The row count before compression is in the compression stats tables.
We have to account for duplicate chunks for the distributed case .

Comment (pmwkaa) [reply sql/size_utils.sql:427]: We haven't done this before, maybe introduce compression support in separate PR in that case? 

Not really sure about the duplicate chunks case, can you specify? I am not sure we support native replication yet

Comment (svenklemm) [reply sql/size_utils.sql:427]: actually with brians patch applied this will get correct stats for compressed and uncompressed chunks

**Comment by pmwkaa** [comment]
Will try different approach after discussion with Erik

**Comment by pmwkaa** [comment]
@gayyappan @svenklemm  

Apparently `hypertable_approximate_row_count()` should work out of the box right after `ANALYZE` being executed on distributed hypertable. We support `ANALYZE` in a way that it will fetch stats from remote nodes and apply them locally on the access node. There is even native replication support already, because access node does not create additional chunks, but data nodes keeps the duplicates. In case of replicated chunk last stats for the same chunk will be applied (stats are essentially should be the same on copies).

So instead of previous implementation, I've added tests cases instead.

**Review by k-rus — COMMENTED** [review]
The PR adds test for the function. Thus the commit message should reflect what the commits do. Current message is misleading as there is only test code.

**Review by erimatnor — APPROVED** [review]

**Pull request merged**